feature/google/src/main/java/com/revenuecat/purchases/google/storeProductConversions.kt

```diff
@@ -59,9 +63,17 @@ fun List<ProductDetails>.toStoreProducts(): List<StoreProduct> {
             val offerDetailsForBasePlan = offerDetailsBySubPeriod[basePlanBillingPeriod] ?: emptyList()
 
             productDetails.toStoreProduct(offerDetailsForBasePlan).let {
-                storeProducts.add(it)
+                it?.let { product -> storeProducts.add(product) } ?: log(
```

Comment:
i think we can avoid the inner `it?.let` if we make the outer `let`s into `?.let`s...

this might not be perfect, but:

```basePlans.takeUnless { it.isEmpty() }?.forEach { basePlan ->
            val basePlanBillingPeriod = basePlan.subscriptionBillingPeriod
            val offerDetailsForBasePlan = offerDetailsBySubPeriod[basePlanBillingPeriod] ?: emptyList()

            productDetails.toStoreProduct(offerDetailsForBasePlan)?.let { product ->
                storeProducts.add(product)
            } ?: log(
                LogIntent.RC_ERROR, PurchaseStrings.INVALID_PRODUCT_NO_PRICE
                    .format(productDetails.productId)
            )
        } ?: productDetails.toInAppStoreProduct()?.let { product ->
            storeProducts.add(product)
        } ?: log(
            LogIntent.RC_ERROR, PurchaseStrings.INVALID_PRODUCT_NO_PRICE
                .format(productDetails.productId)
        )